### PR TITLE
youtube-viewer 3.5.6 (new formula)

### PR DESCRIPTION
**Pull request by TheKevJames**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/youtube-viewer.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    system "perl", "Build.PL", "--install_base", libexec
+    system "./Build", "install"
+
+    %w[youtube-viewer].each do |cmd|
+      (bin/cmd).write_env_script(libexec/"bin/#{cmd}", :PERL5LIB => ENV["PERL5LIB"])
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/youtube-viewer --version").split(/ /)[2]
```
Comment (javian) [inline Formula/youtube-viewer.rb]: Is there any way to test the functionality of this software rather than just printing the version number ?

**Review by javian — CHANGES_REQUESTED** [review]

File: Formula/youtube-viewer.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/youtube-viewer.rb]: Is this tool only interactive?

Comment (TheKevJames) [reply Formula/youtube-viewer.rb]: @javian @SMillerDev is this good as-is or do I need to make changes here to get this accepted?

Comment (SMillerDev) [reply Formula/youtube-viewer.rb]: There needs to be some test. Maybe you could do it like the apache tests where you run and then kill it and check some output?

Comment (TheKevJames) [reply Formula/youtube-viewer.rb]: Is... that not what I'm already doing? Run `youtube-viewer --version`, let it produce output then quit, capture the output and ensure it has the correct value.

Comment (JuPlutonic) [reply Formula/youtube-viewer.rb]: please test somebody on Ubuntu >= 19.04 I have Ubuntu 19.04 and got:
```
make
Last 15 lines from /home/salessionato/.cache/Homebrew/Logs/youtube-viewer/17.make:
Reading MacThai (MacThai)
Reading MacTurkish (MacTurkish)
Reading MacUkrainian (MacUkrainian)
Reading nextstep (nextstep)
Reading viscii (viscii)
"/usr/bin/perl" "/usr/share/perl/5.28.1/ExtUtils/xsubpp"  -nolinenumbers -typemap '/usr/share/perl/5.28/ExtUtils/typemap'  Byte.xs > Byte.xsc
x86_64-linux-gnu-gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=declaration-after-statement -Wpointer-sign -O2 -g   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.28/CORE"   byte_t.c
mv Byte.xsc Byte.c
x86_64-linux-gnu-gcc -c -I./Encode  -I../Encode -D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=declaration-after-statement -Wpointer-sign -O2 -g   -DVERSION=\"2.04\" -DXS_VERSION=\"2.04\" -fPIC "-I/usr/lib/x86_64-linux-gnu/perl/5.28/CORE"   Byte.c
Byte.c:10:20: fatal error: EXTERN.h: No such file or directory
compilation terminated.
make[1]: *** [Makefile:322: Byte.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/tmp/youtube-viewer--Encode-20191125-5568-1ipxfa9/Encode-3.01/Byte'
make: *** [Makefile:649: subdirs] Error 2
```

- which -a perl shows first lonuxbrew's perl

- ~/.cpan removed

- local:lib in $HOME/perl5 removed and recreated with source ~/.bashrc

File: Formula/youtube-viewer.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  # runtime deps, see: https://github.com/trizen/youtube-viewer/blob/master/META.json#L34-L53
+  resource "Data::Dump" do
+    url "https://cpan.metacpan.org/authors/id/G/GA/GAAS/Data-Dump-1.23.tar.gz"
+    sha256 "af53b05ef1387b4cab4427e6789179283e4f0da8cf036e8db516ddb344512b65"
+  end
+
+  resource "Encode" do
+    url "https://cpan.metacpan.org/authors/id/D/DA/DANKOGAI/Encode-3.01.tar.gz"
+    sha256 "d4555f570491648dbbd602bce7966672834b4c8f45acaa6757de474fca3a4d87"
+  end
```
Comment (JuPlutonic) [inline Formula/youtube-viewer.rb]:
```ruby
end unless OS.linux?
```
I concern what all Linux users (not only Ubuntu-users) have Encode installed.
But it is just suggestion I don't understand anything at all about ~/.cpan ~/perl5 or local:lib, after digging homebrew's perl, plenv, some reading (about Pod)... The next error I got after skipping Encode is XSAWYERX/PathTools.

File: Formula/youtube-viewer.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Formula/youtube-viewer.rb]: @JuPlutonic this repository doesn't support Linux. file an issue on linuxbrew-core.

Comment (TheKevJames) [reply Formula/youtube-viewer.rb]: @MikeMcQuaid sorry, could you clarify what action you want here? Do I need to create a companion PR on `linuxbrew-core` adding this formula there, as well, before we can get this one merged here?

Comment (Bo98) [reply Formula/youtube-viewer.rb]: He was responding to a comment by someone else that has since been deleted.

Comment (TheKevJames) [reply Formula/youtube-viewer.rb]: Ah, my bad -- thanks @Bo98 !

Comment (JuPlutonic) [reply Formula/youtube-viewer.rb]: No, youtube-viewer will misbehave if something happen to perl/SSL-certificates or to Youtube's servers (and their GET queries and hypermedia, if presented, seems to me are unchangeable).

Do the perl testing is not needed at all in this formula and do smth like opening the app interactively and producing queries is nonsense -
 we haven't have instruments like `brew test formula --listen_port=8080 --listen_url=0.0.0.0 --query_type=get`

> There needs to be some test. Maybe you could do it like the apache tests where you run and then kill it and check some output?

Comment (MikeMcQuaid) [reply Formula/youtube-viewer.rb]: > youtube-viewer will misbehave if something happen to perl/SSL-certificates(I mean the y-v dependencies) or to Youtube's servers (and their GET queries and hypermedia, if presented, seems to me are unchangeable).

This is desirable: we want the test to fail in this case.

Comment (TheKevJames) [reply Formula/youtube-viewer.rb]: > youtube-viewer [tests] will misbehave if something happen to perl/SSL-certificates(I mean the y-v dependencies)

That seems desireable...

> or to Youtube's servers (and their GET queries and hypermedia, if presented, seems to me are unchangeable)

...but that does not; that'll be very flaky tests, most likely. Do we need to build a mock YouTube server for this?

Comment (MikeMcQuaid) [reply Formula/youtube-viewer.rb]: > Do we need to build a mock YouTube server for this?

No. 

> that'll be very flaky tests, most likely

I disagree. YouTube's uptime is extremely high. We would rather tests that are periodically flaky (and can be rerun) to those that do not exercise the actual functionality of the application.

**Comment by JuPlutonic** [comment]
@TheKevJames, @javian, so thanx for your valuable comments. We in Homebrew need to realize this checks/tests with vcr + webmock gems for such …

**Pull request closed without merging**